src/types/index.ts

```diff
@@ -135,6 +135,14 @@ type Destination = {
   IsConnectionEnabled?: boolean;
 };
 
+type Connection = {
+  sourceId: string;
+  destinationId: string;
+  enabled: boolean;
+  config: Record<string, unknown>;
+  processorEnabled?: boolean;
```

Comment:
why processorEnabled is part of this object ? this won't change at connection level

Reply:
not sure what is this flag means, this is in the rudder-server contract so we used it, we are not using this anywhere

Reply:
@ktgowtham can you explain what is this field means?

Reply:
It's set by config-be [here](https://github.com/rudderlabs/rudder-config-backend/blob/79e6b78493f6b671c8d134cf5c70d9d1d0b0f47d/src/services/workspaceConfig/converter.ts#L336)
This field will be false if it's a device mode connection. processor skips this if it's false.
sry for delayed response.

Reply:
In that case, request reaching transformer will always have processorEnabled as true regardless of destination being on processorTransform/routerTransform